rewrite-core/src/test/java/org/openrewrite/config/YamlResourceLoaderTest.java

```diff
@@ -275,4 +275,81 @@ void declarativeAttributionIncludesRecipeListContributors() {
+              name: test.ChangeTextToHello
+              displayName: Change text to hello
+              recipeList:
+                  - org.openrewrite.text.ChangeText:
+                      toText: Hello!
+              """.getBytes()
+          ), URI.create("rewrite.yml"), new Properties()))
+          .load(new YamlResourceLoader(new ByteArrayInputStream(
+            //language=yml
+            """
+              type: specs.openrewrite.org/v1beta/example
+              recipeName: test.ChangeTextToHello
+              examples:
+                - description: "Change World to Hello in a text file"
+                  before: "World"
+                  after: "Hello!"
+                  language: "text"
+                - description: "Change World to Hello in a java file"
+                  before: |
+                    public class A {
+                        void method() {
+                            System.out.println("World");
+                        }
+                    }
+                  after: |
+                    public class A {
+                        void method() {
+                            System.out.println("Hello!");
+                        }
+                    }
+                  language: "java"
+              """.getBytes()
+          ), URI.create("attribution/test.ChangeTextToHello.yml"), new Properties()))
+          .build();
+
+        Collection<Recipe> recipes = env.listRecipes();
+        assertThat(recipes).hasSize(1);
+        Recipe recipe = recipes.iterator().next();
+        List<RecipeExample> examples = recipe.getExamples();
+        assertThat(examples).hasSize(2);
+        RecipeExample example0 = examples.get(0);
+        assertThat(example0.getDescription()).isEqualTo("Change World to Hello in a text file");
+        assertThat(example0.getBefore()).isEqualTo("World");
+        assertThat(example0.getAfter()).isEqualTo("Hello!");
+        assertThat(example0.getLanguage()).isEqualTo("text");
+        RecipeExample example1 = examples.get(1);
+        assertThat(example1.getDescription()).isEqualTo("Change World to Hello in a java file");
+        assertThat(example1.getBefore()).isEqualTo("""
+          public class A {
+              void method() {
+                  System.out.println("World");
+              }
+          }
+          """);
+        assertThat(example1.getAfter()).isEqualTo("""
+          public class A {
+              void method() {
+                  System.out.println("Hello!");
+              }
+          }
+          """);
+        assertThat(example1.getLanguage()).isEqualTo("java");
+
+        Collection<RecipeDescriptor> recipeDescriptors = env.listRecipeDescriptors();
+        assertThat(recipeDescriptors).hasSize(1);
+        RecipeDescriptor descriptor = recipeDescriptors.iterator().next();
+        List<RecipeExample> descriptorExamples = descriptor.getExamples();
+        assertThat(descriptorExamples).containsExactlyElementsOf(examples);
```

Comment:
I always like it if there's a bit more structure and indentation to my assertions; makes it clearer what the nesting output is like. How'd you feel about:
```suggestion
        Collection<Recipe> recipes = env.listRecipes();
        assertThat(recipes).singleElement().satisfies(r -> {
            assertThat(r.getExamples()).hasSize(2);
            assertThat(r.getExamples()).first().satisfies(e -> {
                assertThat(e.getDescription()).isEqualTo("Change World to Hello in a text file");
                assertThat(e.getBefore()).isEqualTo("World");
                assertThat(e.getAfter()).isEqualTo("Hello!");
                assertThat(e.getLanguage()).isEqualTo("text");
            });
            assertThat(r.getExamples().get(1)).satisfies(e -> {
                assertThat(e.getDescription()).isEqualTo("Change World to Hello in a java file");
                assertThat(e.getBefore()).isEqualTo("""
                  public class A {
                      void method() {
                          System.out.println("World");
                      }
                  }
                  """);
                assertThat(e.getAfter()).isEqualTo("""
                  public class A {
                      void method() {
                          System.out.println("Hello!");
                      }
                  }
                  """);
                assertThat(e.getLanguage()).isEqualTo("java");
            });
        });

        Collection<RecipeDescriptor> recipeDescriptors = env.listRecipeDescriptors();
        assertThat(recipeDescriptors).singleElement().satisfies(descriptor -> {
            List<RecipeExample> descriptorExamples = descriptor.getExamples();
            assertThat(descriptorExamples).containsExactlyElementsOf(recipes.iterator().next().getExamples());
        });
```